Update airflow-api.md for Software (#635)
astronomer/docs: software/airflow-api.md

## Step 2: Make an Airflow REST API Request

> Now that you've created a Service Account, you're free to generate REST API requests to any supported endpoints in Airflow's [Rest API Reference](https://airflow.apache.org/docs/stable/rest-api-ref.html) via the following base URL:

Reviewer: ```suggestion
With the information from Step 1, you can now run `GET` or `POST` requests to any supported endpoints in Airflow's [Rest API Reference](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html) via the following base URL:
```

Reviewer: Making language consistent to Astro

```
https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-NAME>/airflow/api/v1
```

> In the examples below, you'll replace `<BASE-DOMAIN>` (e.g. `mycompany.astronomer.io`) and `<DEPLOYMENT-NAME>` (e.g. `galactic-stars-1234`) with your own.

Reviewer: ```suggestion
In the examples below, you'll replace the following values with your own:

- `<BASE-DOMAIN>`: The base domain for your organization on Astronomer Software. For example: `mycompany.astronomer.io`.
- `<DEPLOYMENT-RELEASE-NAME>`: The release name of your Deployment. For example: `galactic-stars-1234`.
- `<API-KEY>`: The API key for your Deployment Service Account.
```

Reviewer: We refer to this value as `release-name` all over our docs, so I think let's just keep it that way for now. Added this list of placeholder values here for now and can delete the repetitive list below?

You can make requests via the method of your choosing. Below, we'll share example requests via cURL and Python to Airflow's "Trigger DAG" and "Get all Pools" endpoints. In all cases, your request will have the same permissions as the role of the Service Account you created on Astronomer.

### Trigger DAG

To trigger a DAG, execute a POST request to the [dagRuns endpoint](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_dag_run) of the Airflow REST API:

```
POST /dags/<dag-id>/dagRuns
```

> #### cURL

Reviewer: ```suggestion
#### cURL

```

```

> curl -v -X POST https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-NAME>/airflow/api/v1/dags/<DAG-ID>/dagRuns \

Reviewer: ```suggestion
curl -v -X POST https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-RELEASE-NAME>/airflow/api/v1/dags/<DAG-ID>/dagRuns \
```

  -H 'Authorization: <API-KEY>' \
  -H 'Cache-Control: no-cache' \
  -H 'content-type: application/json' -d '{}'
```

> #### Python

Reviewer: ```suggestion
#### Python

```

```python
import requests

token = "<API-KEY>"
base_domain = "<BASE-DOMAIN>"

> deployment_name = "<DEPLOYMENT-NAME>"

Reviewer: ```suggestion
deployment_name = "<DEPLOYMENT-RELEASE-NAME>"
```

resp = requests.post(
    url=f"https://deployments.{base_domain}/{deployment_name}/airflow/api/v1/dags/example_dag/dagRuns",
    headers={"Authorization": token, "Content-Type": "application/json"},
    data='{}'
)
print(resp.json())
# {'conf': {}, 'dag_id': 'example_dag', 'dag_run_id': 'manual__2022-04-26T21:57:23.572567+00:00', 'end_date': None, 'execution_date': '2022-04-26T21:57:23.572567+00:00', 'external_trigger': True, 'logical_date': '2022-04-26T21:57:23.572567+00:00', 'start_date': None, 'state': 'queued'}
```

> To run this, replace the following placeholder values:
> 

Reviewer: ```suggestion
```

> - `<BASE-DOMAIN>`: Use your base domain name, i.e. the domain name used when authenticating to the `astro` cli
> - `<DEPLOYMENT-NAME>`: Use the name of your deployment release name

Reviewer: ```suggestion
```

- `<DAG-ID>`: Name of your DAG (_case-sensitive_)

> - `<API-KEY>`: API Key from your Service Account

Reviewer: ```suggestion
```

> This will trigger a DAG run for your desired DAG with a `execution_date` value of `NOW()`, which is equivalent to clicking the "Play" button in the main "DAGs" view of the Airflow UI.

Reviewer: The python example above specifies `'execution_date': '2022-04-26T21:57:23.572567+00:00'` right? I'd say tweak that to be the equivalent of `NOW()`  and keep the description, and then we can do the specify execution date thing below

## Step 2: Make an Airflow REST API Request

With the information from Step 1, you can now run `GET` or `POST` requests to any supported endpoints in Airflow's [Rest API Reference](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html) via the following base URL:

```
https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-NAME>/airflow/api/v1
```

> In the examples below, you'll replace the following values with your own:

Reviewer 2: ```suggestion
## Example API Calls

In the following examples, you'll replace the following values with your own:
```

- `<BASE-DOMAIN>`: The base domain for your organization on Astronomer Software. For example: `mycompany.astronomer.io`.
- `<DEPLOYMENT-RELEASE-NAME>`: The release name of your Deployment. For example: `galactic-stars-1234`.
- `<API-KEY>`: The API key for your Deployment Service Account.

> You can make requests via the method of your choosing. Below, we'll share example requests via cURL and Python to Airflow's "Trigger DAG" and "Get all Pools" endpoints. In all cases, your request will have the same permissions as the role of the Service Account you created on Astronomer.

Reviewer 2: ```suggestion
The following example calls are made via cURL and Python, but you can make requests via any standard method. In all cases, your request will have the same permissions as the role of the Service Account you created on Astronomer.
```

Here, your request becomes:

```

> curl -v -X POST https://deployments.<AIRFLOW_DOMAIN>/airflow/api/v1/dags/<DAG-ID>/dagRuns \

Reviewer: ```suggestion
curl -v -X POST https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-RELEASE-NAME>/airflow/api/v1/dags/<DAG-ID>/dagRuns \
```

### List Pools

To list all Airflow pools for your Deployment, execute a GET request to the [`pools` endpoint](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#tag/Pool) of the Airflow REST API:

```
GET /pools
```
#### cURL
```

> curl -X GET https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-NAME>/airflow/api/v1/pools \

Reviewer: ```suggestion
curl -X GET https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-RELEASE-NAME>/airflow/api/v1/pools \
```

  -H 'Authorization: <API-KEY>'
```
#### Python
Here, your request would look like this:

```python
import requests

token = "<API-KEY>"
base_domain = "<BASE-DOMAIN>"

> deployment_name = "<DEPLOYMENT-NAME>"

Reviewer: ```suggestion
deployment_name = "<DEPLOYMENT-RELEASE-NAME>"
```

resp = requests.get(
    url=f"https://deployments.{base_domain}/{deployment_name}/airflow/api/v1/pools",
    headers={"Authorization": token, "Content-Type": "application/json"},
    data='{}'
)
print(resp.json())
# {'pools': [{'name': 'default_pool', 'occupied_slots': 0, 'open_slots': 128, 'queued_slots': 0, 'running_slots': 0, 'slots': 128}], 'total_entries': 1}
```

> To run this, replace the following placeholder values:
> 

Reviewer: ```suggestion
```

> - `<API-KEY>`: API Key from your Service Account
> - `<BASE-DOMAIN>`: Use your base domain name, i.e. the domain name used when authenticating to the `astro` cli
> - `<DEPLOYMENT-NAME>`: Use the name of your deployment release name

Reviewer: ```suggestion
```

> ## A Note on Airflow 2.x Stable REST API

Reviewer: ```suggestion
## A Note on Airflow 2 Stable REST API
```

### List Pools

To list all Airflow pools for your Deployment, execute a GET request to the [`pools` endpoint](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#tag/Pool) of the Airflow REST API:

```
GET /pools
```
#### cURL
```
curl -X GET https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-RELEASE-NAME>/airflow/api/v1/pools \
  -H 'Authorization: <API-KEY>'
```
#### Python
Here, your request would look like this:

```python
import requests

token = "<API-KEY>"
base_domain = "<BASE-DOMAIN>"
deployment_name = "<DEPLOYMENT-RELEASE-NAME>"
resp = requests.get(
    url=f"https://deployments.{base_domain}/{deployment_name}/airflow/api/v1/pools",
    headers={"Authorization": token, "Content-Type": "application/json"},
    data='{}'
)
print(resp.json())
# {'pools': [{'name': 'default_pool', 'occupied_slots': 0, 'open_slots': 128, 'queued_slots': 0, 'running_slots': 0, 'slots': 128}], 'total_entries': 1}
```

To run this, replace the following placeholder values:

> ## A Note on Airflow 2 Stable REST API
> 
> ### What's new

Reviewer 2: ```suggestion
## Notes on Airflow 2.0's Stable REST API
```

```
curl -X GET \

>   https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-NAME>/api/v1/config \

Reviewer: ```suggestion
  https://deployments.<BASE-DOMAIN>/<DEPLOYMENT-RELEASE-NAME>/api/v1/config \
```

  -H 'Authorization: <API-KEY>' \
  -H 'Cache-Control: no-cache'
```

> To run this, update the following placeholder values:
> 

Reviewer: ```suggestion
```

> - `<BASE-DOMAIN>`: Use your base domain name, i.e. the domain name used when authenticating to the `astro` cli
> - `<DEPLOYMENT-NAME>`: Use the name of your deployment release name
> - `<API-KEY>`: API Key from your Service Account

Reviewer: ```suggestion
```
